Android releasing (#1178)
Azure/azure-sdk-for-android: sdk/communication/azure-communication-chat/CHANGELOG.md

# Release History
## 2.0.0 (2022-08-26)
### Bugs Fixed

> - Push notification registration renewal not guaranteed to happen when Contoso APP is idle or closed.

Reviewer: Are we saying this is no longer happening? The wording is confusing and could make people think the renewal is not guaranteed to happen when in fact, it is.
```suggestion
- Push notification registration renewal is now guaranteed to happen when an app is idle or closed.
```

### Breaking Changes

> - Introduced WorkerManager library which breaks push notification for existing Contoso APPs. (Check https://docs.microsoft.com/azure/communication-services/tutorials/chat-android-push-notification)

Reviewer: ```suggestion
- Introduced the use of WorkManager, which now requires additional configuration in apps that were already using the push notification feature. Check the [push notifications documentation](https://docs.microsoft.com/azure/communication-services/tutorials/chat-android-push-notification) for more information.
```

### Other Changes
- Implement internal key management library to securely persist all the registration secret keys which are not expired.

> - Increase minSDKVersion from 21 to 23.

Reviewer: Not sure if this should be under the breaking changes section, as it can technically be "reversed". Any thoughts @JonathanGiles?

Author: I think "Increase minSDKVersion from 21 to 23" could be put under breaking changes. Because it is breaking change in this release (2.0.0). When we do a reverse, it would be done in another release like 2.1.0. 

Additionally, put it under breaking change section is more obvious for users. They are likely to ignore the "Other changes" section.

Reviewer: Sounds good, let's do it then.

- Reduce the expiration time of each push notification registration from 25 hours to 45 minutes.

> - Add deprecate annotation to startPushNotifications API with errorHandler as a parameter.

Reviewer: ```suggestion
- Add `@Deprecated` annotation to `startPushNotifications` API with `errorHandler` as a parameter.
```

> - Create new API errorHandler without errorHandler as a parameter.

Reviewer: We should probably put this one under `### Features Added` like we've done with other new APIs.
